### PR TITLE
Attempt to get libffi wired up to build fiddle

### PR DESCRIPTION
**Pull request by jd-stripe**
### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jd-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_PahVAx8ZJDBvty
→ https://go/builds/bui_PahWHOuWN5Ncr8
→ https://go/builds/bui_PahWvKx5pn4Jau

**Comment by jd-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_Pahac7yDIFubab
→ https://go/builds/bui_PahaM4Jy4bdGV6
→ https://go/builds/bui_Pahaema9OsSd1m

File: third_party/ruby/build-ruby.bzl [hunk 1]
```diff
@@ -163,7 +168,7 @@ cp "$out_dir/bin/bundle" "$out_dir/bin/bundler"
 
 popd > /dev/null
 
-rm -rf "$build_dir"
+# rm -rf "$build_dir"
```
Comment (fable-stripe) [inline third_party/ruby/build-ruby.bzl:171]: do we sitll need this?

File: third_party/ruby/ldflags.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git ext/fiddle/extconf.rb ext/fiddle/extconf.rb
+index 2d85b3eea5..eb03d3bfa6 100644
+--- ext/fiddle/extconf.rb
++++ ext/fiddle/extconf.rb
+@@ -96,7 +96,7 @@ def enable_debug_build_flag(flags)
+
+   FileUtils.mkdir_p(libffi.dir)
+   libffi.opt = CONFIG['configure_args'][/'(-C)'/, 1]
+-  libffi.ldflags = RbConfig.expand("$(LDFLAGS) #{libpathflag([relative_from($topdir, "..")])} #{$LIBRUBYARG}".dup)
++  libffi.ldflags = RbConfig.expand("$(LDFLAGS) #{libpathflag([relative_from($topdir, "..")])}".dup)
```
Comment (fable-stripe) [inline third_party/ruby/ldflags.patch:10]: what does this change?

**Review by fable-stripe — CHANGES_REQUESTED** [review]

**Comment by fable-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_PlxLg0bli12ju0
→ https://go/builds/bui_PlxLOmtfubyIZY
→ https://go/builds/bui_PlxLX4ELb9gbju

**Comment by fable-stripe** [comment]
Stripes can see https://cibot.corp.stripe.com/build_groups/bg_PlxfRlI3wrggMN

tested that fiddle is functional.
```
:hamburger: fable@fable-doing-ruby-crimes:/pay/src/sorbet$ cat hi.rb
require 'fiddle'


libm = Fiddle.dlopen('./libfoo.so')

floor = Fiddle::Function.new(
  libm['add1'],
  [Fiddle::TYPE_INT],
  Fiddle::TYPE_INT
)

puts floor.call(3)

:hamburger: fable@fable:/pay/src/sorbet$ bazel-bin/external/sorbet_ruby_3_3/ruby hi.rb 
4
```

**Pull request merged**